presto-main/src/main/java/com/facebook/presto/cost/AggregationStatsRule.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.plan.AggregationNode;
+import com.facebook.presto.sql.planner.plan.AggregationNode.Aggregation;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.util.Objects.requireNonNull;
+
+public class AggregationStatsRule
+        implements ComposableStatsCalculator.Rule
+{
+    @Override
+    public Optional<PlanNodeStatsEstimate> calculate(PlanNode node, Lookup lookup, Session session, Map<Symbol, Type> types)
+    {
+        if (!(node instanceof AggregationNode)) {
+            return Optional.empty();
+        }
+        AggregationNode aggregationNode = (AggregationNode) node;
+
+        if (aggregationNode.getGroupingSets().size() != 1) {
+            return Optional.empty();
+        }
+
+        PlanNodeStatsEstimate sourceStats = lookup.getStats(aggregationNode.getSource(), session, types);
+
+        List<Symbol> groupBySymbols = getOnlyElement(aggregationNode.getGroupingSets());
+
+        if (!sourceStats.getSymbolsWithKnownStatistics().containsAll(groupBySymbols)) {
```

Comment:
Probably it can be handled by checking at the bottom of method that `rowsCount` is `NaN` and return UNKNOWN_STATS in such case.